core/src/main/java/tc/oc/pgm/stats/StatsModule.java

```diff
@@ -0,0 +1,56 @@
+package tc.oc.pgm.stats;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Color;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.math.Formula;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLFluentParser;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class StatsModule implements MapModule<StatsMatchModule> {
+  private final List<StatType.OfFormula> formulaStats;
+
+  private StatsModule(List<StatType.OfFormula> formulaStats) {
+    this.formulaStats = formulaStats;
+  }
+
+  @Override
+  public @Nullable StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new StatsMatchModule(match, formulaStats);
+  }
+
+  public static class Factory implements MapModuleFactory<StatsModule> {
```

Comment:
Also one thing i forgot: you need to declare a dependency on the variables module, as stats now need to load *after* variables, so that formulas can work well.
By pure coincidence this may work without the explicit dependency, but it is fragile and may break at any point in the future.

overwrite the `getWeakDependencies` method and return a `List.of(VariablesModule.class);` (you can look at ActionModule for an example

Reply:
Will add